app/controllers/devise_token_auth/sessions_controller.rb

```diff
@@ -34,13 +34,19 @@ def create
         yield @resource if block_given?
 
         render_create_success
-      elsif @resource && !(!@resource.respond_to?(:active_for_authentication?) || @resource.active_for_authentication?)
+      elsif @resource && !Devise.paranoid && !(!@resource.respond_to?(:active_for_authentication?) || @resource.active_for_authentication?)
         if @resource.respond_to?(:locked_at) && @resource.locked_at
           render_create_error_account_locked
         else
           render_create_error_not_confirmed
         end
       else
+        # In order to avoid timing attacks in paranoid mode, we want the password hash to be
```

Comment:
can you move this new code and explanation to another method so doesn't make bigger than it's?

Reply:
Hi, thanks for the review :)
I moved the code in a new method and also took the liberty to refactor the beginning the create method a bit. Hope that's fine.

Reply:
thank you!